### PR TITLE
Improved fields dependence (show/hide) in backend's config section

### PR DESCRIPTION
**Pull request by luigifab**
### Description

This fix display of customized elements when a field depend on another one.

### Manual testing scenarios

First, update _app/code/core/Mage/Page/etc/system.xml_ like this:
```diff
diff --git a/app/code/core/Mage/Page/etc/system.xml b/app/code/core/Mage/Page/etc/system.xml
index 635353f637..88c1aba8c7 100644
--- a/app/code/core/Mage/Page/etc/system.xml
+++ b/app/code/core/Mage/Page/etc/system.xml
@@ -37,6 +37,15 @@
                     <show_in_website>1</show_in_website>
                     <show_in_store>1</show_in_store>
                     <fields>
+                        <yesno>
+                            <label>yesno</label>
+                            <frontend_type>select</frontend_type>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <sort_order>4</sort_order>
+                            <show_in_default>1</show_in_default>
+                            <show_in_website>1</show_in_website>
+                            <show_in_store>1</show_in_store>
+                        </yesno>
                         <shortcut_icon translate="label comment">
                             <label>Favicon Icon</label>
                             <comment>Allowed file types: ICO, PNG, GIF, JPG, JPEG, APNG. Not all browsers support all these formats!</comment>
@@ -47,6 +56,7 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
+                            <depends><yesno>1</yesno></depends>
                         </shortcut_icon>
                         <default_title translate="label">
                             <label>Default Title</label>
```

Go to _System / Configuration / Design / HTML head_:

image

Here, there are no problems with the favicon, when the select is yes or no.

But... if you customize the element and add some html tags, like for example by adding in _app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Image.php_:

```php
    public function getElementHtml() {
        return '<div style="background-color:yellow;">'.parent::getElementHtml().'</div>';
    }
```
No:
image
Yes:
image

Apply the PR, then try again No:
image

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: js/mage/adminhtml/form.js [hunk 1]
```diff
@@ -563,22 +561,22 @@ FormElementDependenceController.prototype = {
         // toggle target row
         if (shouldShowUp) {
             var currentConfig = this._config;
-            $(idTo).up(this._config.levels_up).select('input', 'select', 'td').each(function (item) {
```
Comment (kiatng) [inline js/mage/adminhtml/form.js:566]: This breaks BC for custom code that uses `this._config.levels_up`, although I do not know of any custom code that uses it,

Comment (luigifab) [reply js/mage/adminhtml/form.js:565]: Perhaps I can check if `levels_up` exist and use it if it's true.

Comment (luigifab) [reply js/mage/adminhtml/form.js:539]: Not sure if I must to add the check or not.

File: js/mage/adminhtml/form.js [hunk 2]
```diff
@@ -512,16 +510,21 @@ FormElementDependenceController.prototype = {
      */
     trackChange : function(e, idTo, valuesFrom)
     {
-        let upLevels = this._config.levels_up;
-        let ele;
-        if (!$(idTo)) {
+        var ele = $(idTo), cnf = this._config;;
```
Comment (fballiano) [inline js/mage/adminhtml/form.js:513]:
```suggestion
        var ele = document.getElementById(idTo), cnf = this._config;;
```

File: js/mage/adminhtml/form.js [hunk 3]
```diff
@@ -512,16 +510,21 @@ FormElementDependenceController.prototype = {
...
             if (!ele) {
                 return;
             }
         } else {
-            ele = $(idTo).up(upLevels);
+            var closest = cnf.levels_up; // @deprecated
+            if ((typeof closest == 'number') && (closest > 1)) {
+                ele = ele.up(closest);
+            }
+            else {
```
Comment (fballiano) [inline js/mage/adminhtml/form.js:525]:
```suggestion
            } else {
```

**Review by fballiano — DISMISSED** [review]
tested and I couldn't find anything not working

**Review by empiricompany — APPROVED** [review]

File: js/mage/adminhtml/form.js [hunk 4]
```diff
@@ -512,16 +510,20 @@ FormElementDependenceController.prototype = {
      */
     trackChange : function(e, idTo, valuesFrom)
     {
-        let upLevels = this._config.levels_up;
-        let ele;
-        if (!$(idTo)) {
+        var ele = document.getElementById(idTo), cnf = this._config;;
```
Comment (kiatng) [inline js/mage/adminhtml/form.js:513]:
```suggestion
        let ele = document.getElementById(idTo), cnf = this._config;
```

Comment (kiatng) [reply js/mage/adminhtml/form.js:513]: I think `cnf` should be renamed to `config`.

**Review by kiatng — APPROVED** [review]
Tested. I removed an extra semicolon.

**Review by fballiano — APPROVED** [review]

**Pull request merged**